tracer/src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -464,13 +467,13 @@ DdwafObjectStruct GetStringObject(string? keyForObject, string value)
 
                     if (enumerable is IList { Count: { } count } listInstance)
                     {
-                        if (count > WafConstants.MaxContainerSize)
+                        if (applySafetyLimits && count > WafConstants.MaxContainerSize)
                         {
                             Log.Warning<int>("EncodeList: list too long, it will be truncated, MaxMapOrArrayLength {MaxMapOrArrayLength}", WafConstants.MaxContainerSize);
                             break;
```

Comment:
I think we dont want to break here, otherwise we dont encode the list at all (?)  I think that's why `TestArrayLengtch(257) doesnt pass 🤔